### PR TITLE
Fix WPML clearing all languages cache

### LINKED ISSUE #6214: Avoid clearing cache once per language
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version
- Used the search feature to ensure that the bug hasn’t been reported before

**Describe the bug**
WPML Team reported that when translating a single post we're clearing the cache once per language. It's possible to do it only once to avoid resources problems.

**To Reproduce**
Included in https://github.com/wp-media/wp-rocket/issues/6214#issuecomment-1757574484

**Expected behavior**
Included in https://github.com/wp-media/wp-rocket/issues/6214#issuecomment-1757574484

**Acceptance Criteria (for WP Media team use only)**
- We should launch `rocket_clean_domain` only once, not once per language
- No regression at clearing the translated posts
- No regression at clearing the specific languages

### PR DESCRIPTION
**Pull request by wordpressfan**
## Description

Here we introduced a new filter that runs after clearing cache and used in preload as in the issue's grooming.

Fixes #6214

## Type of change

*Please delete options that are not relevant.*

- Bug fix (non-breaking change which fixes an issue).
- Enhancement (non-breaking change which improves an existing functionality).

## Is the solution different from the one proposed during the grooming?

Exactly as it's groomed.

# Checklists

## Generic development checklist

- [x] My code follows the style guidelines of this project, with adapted comments and without new warnings.
- [x] I have added unit and integration tests that prove my fix is effective or that my feature works.
- [x] The CI passes locally with my changes (including unit tests, integration tests, linter).
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] If applicable, I have made corresponding changes to the documentation. *Provide a link to the documentation.*

## Test summary

- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [ ] I validated all Acceptance Criteria of the related issues. (If applicable, provide proof).
- [ ] I validated all test plan the QA Review asked me to.

*If not, detail what you could not test.*

*Please describe any additional tests you performed.*

### CODE REVIEW

File: inc/functions/files.php [hunk 1]
```diff
@@ -876,6 +876,16 @@ function rocket_clean_domain( $lang = '', $filesystem = null ) {
 		do_action( 'after_rocket_clean_domain', $root, $lang, $url ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals
 	}
 
+	/**
+	 * Fires after all cache files was deleted.
+	 *
+	 * @since 3.15.5
+	 *
+	 * @param string $lang The current lang to purge.
+	 * @param string $url  All urls to clean.
```
Comment (nicomollet) [inline inc/functions/files.php]: Is `$urls` not an array?

**Review by nicomollet — COMMENTED** [review]
The doc to fix?

**Review by jeawhanlee — APPROVED** [review]

**Review by remyperona — COMMENTED** [review]
Did you check if there was other instances of callbacks on `after_rocket_clean_domain` that could be replaced to optimize this?

File: inc/functions/files.php (hunk 1 above)
Comment (wordpressfan) [reply inc/functions/files.php]: Good catch, Done.

**Comment by wordpressfan** [comment]
> Did you check if there was other instances of callbacks on `after_rocket_clean_domain` that could be replaced to optimize this?

That's a good point, I saw some cases but I wasn't sure if we need to do this here and overload this PR or not, but I believe u r correct, let me check them and see.

**Review by remyperona — APPROVED** [review]

**Comment by Mai-Saad** [comment]
@wordpressfan Thanks for the PR. A request is done once now when changing settings, full cache clear from admin, clearing used CSS, changing permalinks  (assuming that all will trigger this function https://github.com/wp-media/wp-rocket/blob/b67069fc7f18be29f73952efb1489ee18004ec80/inc/Engine/Preload/Subscriber.php#L285). However, when we switch theme can see many calls to the same function, can you please check :pray:

**Comment by wordpressfan** [comment]
When we switch theme we are hooking `rocket_clean_domain` callback with three different hooks in the following locations:

**First location:**
https://github.com/wp-media/wp-rocket/blob/afef3b6bc067af18de183a10e337e17eef5707db/inc/common/purge.php#L16

**Second location:**
https://github.com/wp-media/wp-rocket/blob/afef3b6bc067af18de183a10e337e17eef5707db/inc/common/purge.php#L8

**Third and main location:**
https://github.com/wp-media/wp-rocket/blob/afef3b6bc067af18de183a10e337e17eef5707db/inc/common/purge.php#L6

I believe this is another issue.

cc @piotrbak

**Comment by piotrbak** [comment]
@Mai-Saad @wordpressfan Yes, this is different issue. Let's go with separate issue for this one

**Comment by Mai-Saad** [comment]
Working as expected then. (Note, the behavior when switch theme is the same even without WPML)

**Comment by wordpressfan** [comment]
The other issue is created here: https://github.com/wp-media/wp-rocket/issues/6276

**Pull request merged**